File: grpc/examples/python/get-network-parameters.py
#!/usr/bin/python3

"""
Script language: Python3

Talks to:
- Vega node (gRPC)

Apps/Libraries:
- gRPC: Vega-API-client (https://pypi.org/project/Vega-API-client/)
"""

# Note: this file uses smart-tags in comments to section parts of the code to
# show them as snippets in our documentation. They are not necessary to be
# included when creating your own custom code.
#
# Example of smart-tags:
#  __something:
# some code here
# :something__

import helpers

# __import_client:
import vegaapiclient as vac

# Vega gRPC clients for reading/writing data
node_url_grpc = helpers.get_from_env("NODE_URL_GRPC")
data_client = vac.VegaTradingDataClient(node_url_grpc)
# :import_client__

###############################################################################
#                      N E T W O R K   P A R A M E T E R S                    #
###############################################################################

# __get_network_params:
# Request a list of network parameters configured on a Vega network
request = vac.api.trading.NetworkParametersRequest()
network_params = data_client.NetworkParameters(request)
print("Network Parameter:\n{}".format(network_params))
# :get_network_params__

File: grpc/examples/go/get-network-parameters.go
package main

import (
	"context"
	"fmt"

	"github.com/vegaprotocol/api/grpc/clients/go/generated/code.vegaprotocol.io/vega/proto/api"
	"github.com/vegaprotocol/api/grpc/examples/go/helpers"

	"google.golang.org/grpc"
)

func main() {
	nodeURLGrpc := helpers.GetFromEnv("NODE_URL_GRPC")
	conn, err := grpc.Dial(nodeURLGrpc, grpc.WithInsecure())
	if err != nil {
		panic(err)
	}
	defer conn.Close()

	dataClient := api.NewTradingDataServiceClient(conn)

	// __get_network_params:
	// Request a list of network parameters configured on a Vega network
	request := api.NetworkParametersRequest{}
	network, err := dataClient.NetworkParameters(context.Background(), &request)
	if err != nil {
		panic(err)
	}

	fmt.Printf("Network Parameters: %s", network)
	// :get_network_params__
}
